### PR TITLE
Enhamcement/43 use warning on date validation on branches

### LINKED ISSUE #43: Replace exception with warning on the date validation for branches

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #43

### CODE REVIEW

File: src/main/java/com/exasol/validation/GitHubPlatformValidator.java [hunk 1]
```diff
@@ -71,10 +69,20 @@ protected void validateContainsHeader(final ReleaseLetter changes) {
...
+        final String wrongTicketsString = String.join(", ", wrongTickets);
+        if (this.branchContent.isDefaultBranch()) {
             throw new IllegalStateException(
                     "E-RR-VAL-2: Some of the mentioned GitHub issues are not closed or do not exists: "
-                            + String.join(", ", wrongTickets) + ", Please, check the issues numbers in your '"
-                            + changesFile.getFileName() + "' one more time.");
+                            + wrongTicketsString + ", Please, check the issues numbers in your '" + fileName
+                            + "' one more time.");
+        } else {
+            LOGGER.info("Don't forget to close the tickets mentioned in the '" + fileName
```
Comment (jakobbraun) [inline src/main/java/com/exasol/validation/GitHubPlatformValidator.java:84]: in the ticket description a *warning* is requested. Maybe use `LOGGER.warn` instead?

File: src/test/java/com/exasol/validation/GitRepositoryValidatorTest.java [hunk 2]
```diff
@@ -58,7 +58,18 @@ void testValidateChangesInvalidDate() {
         when(changesMock.getReleaseDate()).thenReturn(Optional.of(LocalDate.of(2020, 8, 1)));
         when(changesMock.getBody()).thenReturn(Optional.of("## Features"));
         when(changesMock.getFileName()).thenReturn("file");
-        assertThrows(IllegalStateException.class, () -> this.validator.validateChanges(changesMock, "2.1.0"));
+        assertThrows(IllegalStateException.class, () -> this.validator.validateChanges(changesMock, "2.1.0", true));
```
Comment (jakobbraun) [inline src/test/java/com/exasol/validation/GitRepositoryValidatorTest.java:61]: Maybe also validate the exception message?

File: src/test/java/com/exasol/validation/GitRepositoryValidatorTest.java [hunk 3]
```diff
@@ -58,7 +58,18 @@ void testValidateChangesInvalidDate() {
...
+
+    @Test
+    // [utest->dsn~validate-changes-file-contains-release-date~1]
+    void testValidateChangesInvalidDateWarning() {
+        final ReleaseLetter changesMock = Mockito.mock(ReleaseLetter.class);
+        when(changesMock.getVersionNumber()).thenReturn(Optional.of("2.1.0"));
+        when(changesMock.getReleaseDate()).thenReturn(Optional.of(LocalDate.of(2020, 8, 1)));
+        when(changesMock.getBody()).thenReturn(Optional.of("## Features"));
+        when(changesMock.getFileName()).thenReturn("file");
+        assertDoesNotThrow(() -> this.validator.validateChanges(changesMock, "2.1.0", false));
```
Comment (jakobbraun) [inline src/test/java/com/exasol/validation/GitRepositoryValidatorTest.java:72]: Validate warning
You could inject a Mock logger using reflection and then use Mockito.verfiy to check that the method was called.

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: src/main/java/com/exasol/validation/GitHubPlatformValidator.java (hunk 1 above)
Comment (AnastasiiaSergienko) [reply src/main/java/com/exasol/validation/GitHubPlatformValidator.java:84]: Good idea!

File: src/test/java/com/exasol/validation/GitRepositoryValidatorTest.java (hunk 2 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/validation/GitRepositoryValidatorTest.java]: Added

File: src/test/java/com/exasol/validation/GitRepositoryValidatorTest.java (hunk 3 above)
Comment (AnastasiiaSergienko) [reply src/test/java/com/exasol/validation/GitRepositoryValidatorTest.java:74]: I think there is no need to validate the logger message here. And I don't like to use reflection until it's 100% necessary

Comment (jakobbraun) [reply src/test/java/com/exasol/validation/GitRepositoryValidatorTest.java:74]: ok

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**